server/src/main/java/org/elasticsearch/common/io/stream/StreamOutput.java

```diff
@@ -770,6 +772,22 @@ public final void writeOptionalInstant(@Nullable Instant instant) throws IOExcep
         })
     );
 
+    private static void writeHashMap(StreamOutput o, Map<String, Object> map) throws IOException {
```

Comment:
Similar comment as to the read side. We already have write methods for maps, there should be no need for new methods.